### DISCUSSION TITLE
Disable initial download of whole pdf

### DISCUSSION THREAD

**Discussion by Rameshmariappan** [wojtekmaj/react-pdf#1576] [Q&A]
I want to disable the initial download of the whole pdf, instead, I want to fetch 4-5 pages and if I scroll the page remaining needs to fetch. while searching I found in pdf js I have an option like disableAutoFetch is there any option in react-pdf?
![image](https://github.com/wojtekmaj/react-pdf/assets/56838939/d26d1083-a38f-4697-a731-bb0788be6ce9)

**Comment by wojtekmaj** [comment]
Sure, you should be able to pass this in `options` prop :D

**Reply by Rameshmariappan** [reply]
@wojtekmaj 
I have a PDF stored in my S3 bucket, and I want to display it within my application. The PDF size is approximately 7MB. When I use the URL in the <Document> component, the entire file is downloaded upfront, which can be slow and consume a lot of data. Is there a way to download only a portion of the PDF, like the first 2-3 pages, and then load the remaining pages dynamically as the user scrolls through the document? or is there any way to show progress data like 1 MB completed from 7Mb
![image](https://github.com/wojtekmaj/react-pdf/assets/56838939/5a79baef-a812-4e1a-be46-19b373036336)

**Reply by wojtekmaj** [reply]
PDF.js supports partial download out of the box, so provided that the server is capable of handling byte range requests (and Amazon S3 is), React-PDF will do a partial file download automatically. If it doesn't, chances are the PDF is not prepared in a way that allows partial download or it's too small to be downloaded in parts.
